### PR TITLE
Improved usage instructions in readme file

### PR DESCRIPTION
**Pull request by gennadiylitvinyuk**
|    Q          |   A
|-------------- | ------
| Documentation | yes
### Description

Improved usage instructions
Fixed docker image name in local usage example

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +30,19 @@ Generally speaking, you will use this in combination with the [laminas/laminas-c
...
+Create `.github/workflows/continuous-integration.yml` file with following content:
+
 ```yaml
+name: "Continuous Integration"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - '[0-9]+.[0-9]+.x'
```
Comment (boesing) [inline README.md]: This may not apply to all projects out there.
I'd say remove this line from here to avoid issues in other projects.

**Review by boesing — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (gennadiylitvinyuk) [reply README.md]: How about `on: [push, pull_request]`

File: README.md [hunk 2]
```diff
@@ -155,7 +160,7 @@ The tricks to remember are:
 As an example, if you wanted to run the CS checks under PHP 7.4 using locked dependencies, you could do something like the following:
 
 ```bash
-$ docker run -v $(realpath .):/github/workspace -w=/github/workspace laminas-check-runner:latest '{"php":"7.4","deps":"locked","extensions":[],"ini":["memory_limit=-1"],"command":"./vendor/bin/phpcs"}'
+$ docker run -v $(realpath .):/github/workspace -w=/github/workspace ghcr.io/laminas/laminas-continuous-integration:1  '{"php":"7.4","deps":"locked","extensions":[],"ini":["memory_limit=-1"],"command":"./vendor/bin/phpcs"}'
```
Comment (boesing) [inline README.md]: I am not sure but I guess this was meant to be used locally.
Imho, the `docker build . -t laminas-check-runnter:latest` command is missing somewhere. Then, this line is absolutely okayish.
But I am not sure about this. Maybe @weierophinney can say something about it.

File: README.md [hunk 3]
```diff
@@ -30,7 +30,13 @@ Generally speaking, you will use this in combination with the [laminas/laminas-c
 
 ## Usage
 
+Create `.github/workflows/continuous-integration.yml` file with following content:
+
 ```yaml
+name: "Continuous Integration"
+
+on: [push, pull_request]
```
Comment (weierophinney) [inline README.md]: I should probably explain why I used the following:

```yaml
on:
  pull_request:
  push:
    branches:
      - '[0-9]+.[0-9]+.x'
      - 'refs/pull/*'
      tags:
```

What I found (through a LOT of trial and error), was the following:

- I started with just pull_request. When I did that, additional pushes to the PR branch would not trigger checks.
- When I added "push", without qualifications, I would get TWO runs of checks on opening a PR, one for the PR, and one for the branch. This led to unnecessary builds.
- If I added branches to the mix (as I did) to qualify which branch triggered a build... tags would not build.

Basically, I found `on: [push, pull_request]` was building _too much_, which led me to qualify the values.

If you can point to repos where you have rolled out a workflow with the simplified entry, please link me to it so I can see the build history and make sure it's working the way I would want it to; until then, I'm a bit wary of this change.

Comment (boesing) [reply README.md]: I've thought that `- '[0-9]+.[0-9]+.x'` might be too much for projects which do not follow the branch strategy required by the `laminas-automatic-releases` GHA.

Comment (weierophinney) [reply README.md]: We could demonstrate two variants, then, maybe:

- The one we're using (and link to automatic-releases to detail the _why_ aspect)
- A more general one (and detail issues such as double-runs, if they're still valid)

File: README.md (hunk 2 above)
Comment (weierophinney) [reply README.md]: I originally wrote this based on having built locally, but I've since switched to using images I've pulled off of ghcr.io. That feels more natural as a docker user (pull and use prebuilt images), so I'm good with this change.

Comment (boesing) [reply README.md]: Me too, we could add more details on how to build and use this locally in a future PR 👍🏼

**Comment by Xerkus** [comment]
Rebased to resolve conflicts. Added example with reusable workflow. Added mention of potential duplicate CI runs.

File: README.md [hunk 4]
```diff
@@ -87,6 +95,29 @@ jobs:
           job: ${{ matrix.job }}
 ```
 
+Same can be achieved with our reusable workflow :
```
Comment (weierophinney) [inline README.md]:
```suggestion
The same can be achieved with our reusable workflow:
```

File: README.md [hunk 5]
```diff
@@ -87,6 +95,29 @@ jobs:
...
+    uses: laminas/workflow-continuous-integration/.github/workflows/continuous-integration.yml@1.x
+```
+
+> ### Duplicate CI runs
+>
+> When pull requests are opened from branches in the same repository CI pipeline could run twice for the same commit:
+> once when pushed to a branch and once when PR is opened. Workflow triggers could be adjusted to only run for
+> specific branches or to skip branches using simple glob patterns.
+> See [Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs)
+> for more info.
```
Comment (weierophinney) [inline README.md]:
```suggestion
> When pull requests are opened from branches in the same repository, the CI pipeline could run twice for the same commit: once when pushed to a branch and once when the PR is opened.
> Workflow triggers can be adjusted to only run for specific branches or to skip branches using simple glob patterns.
> See the GitHub documentation for [Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs) for more info.
```

Comment (weierophinney) [reply README.md]: The suggestion above:

- places one sentence per line (this is a style guideline in our docs)
- rephrases slightly for better clarity

**Review by weierophinney — COMMENTED** [review]
Minor suggestions, but changes look good, @Xerkus .

**Comment by Xerkus** [comment]
@gennadiylitvinyuk thank you
